Src/FluentAssertions/Primitives/StringEqualityValidator.cs

```diff
@@ -44,26 +44,14 @@ private string GetMismatchSegmentForStringsOfDifferentLengths()
         {
             int indexOfMismatch = Subject.IndexOfFirstMismatch(Expected, comparisonMode);
 
-            // If there is no difference it means that either
-            // * subject starts with expected or
-            // * expected starts with subject
+            // If there is no difference it means that expected starts with subject and subject is shorter than expected
             if (indexOfMismatch == -1)
             {
-                // If subject is shorter we are sure that expected starts with subject
-                if (Subject.Length < Expected.Length)
-                {
-                    // Subject is shorter so we point at its last character.
-                    // We would like to point at next character as it is the real
-                    // index of first mismatch, but we need to point at character existing in
-                    // subject, so the next best thing is the last subject character.
-                    indexOfMismatch = Math.Max(0, Subject.Length - 1);
-                }
-                else
-                {
-                    // If subject is longer we are sure that subject starts with expected
```

Comment:
What makes you think this scenario is not necessary anymore?

Reply:
It was red in my VS coverage. I then debugged all corresponding tests, e.g. `When_string_start_is_compared_with_string_that_is_longer_it_should_throw()` , they all don't go into the else branch. I even added a new test to be sure. 
Looking at `IndexOfFirstMismatch()`, it compares Subject with Expected by iterating over the characters of Subject. So the only scenario for -1 can be if Subject is shorter than Expected (or equal) and up until the end of Subject the characters are the same, then it stops at the end of Subject and doesn't compare the rest. Otherwise, if Subject is longer than Expected, we get the position of the first differing character.

Reply:
I haven't dug into the logic, but have you tried wrapping your tests in an assertion scope to if the logic handles continuing after the first failure?

The commit/PR adding the logic maybe also has some useful insights.

Reply:
It seems that @chvollm is right. That branch isn't covered by any tests and I can't seem to find an edge case where it fails.

Reply:
Thanks. I spent quite some time on it but was still a bit unsure, so good to see that you can confirm my observation.